src/prefect/_internal/pydantic/v2_schema.py

```diff
@@ -106,7 +106,11 @@ def create_v2_schema(
     model = create_model(
         name_, __config__=model_cfg, __base__=model_base, **model_fields
     )
-    adapter = TypeAdapter(model)
+    try:
+        adapter = TypeAdapter(model)
+    except PydanticUndefinedAnnotation as exc:
+        # in v1 this raises a TypeError, which is handled by parameter_schema
```

Comment:
In v1 `create_schema` raises a type error which is caught by `parameter_schema` to type to `Any`. I am not sure if this is the right way to handle this...

Reply:
these tests were failing without it:
https://github.com/PrefectHQ/prefect/blob/main/tests/test_flows_compat.py#L1-L30

Reply:
hmm - I am also not totally sure about that 🧐

i don't immediately have a better suggestion though